### PR TITLE
JAVA-18131 Upgrade core-java-modules to JDK 11

### PR DESCRIPTION
**Pull request by anastasiosioannidis**

### CODE REVIEW

**Comment by asif-anwar** [comment]
retest this please

**Comment by anastasiosioannidis** [comment]
retest this please

File: core-java-modules/core-java-jvm/pom.xml [hunk 1]
```diff
@@ -64,6 +57,24 @@
         </dependency>
     </dependencies>
 
+<!--    <build>-->
```
Comment (dkapil) [inline core-java-modules/core-java-jvm/pom.xml:60]: can we remove this entirely if not needed? let's not have commented code

File: core-java-modules/core-java-jvm/pom.xml [hunk 2]
```diff
@@ -176,10 +187,12 @@
 
     <properties>
         <javaassist.version>3.27.0-GA</javaassist.version>
-        <sun.tools.version>1.8.0</sun.tools.version>
         <jol-core.version>0.10</jol-core.version>
-        <asm.version>8.0.1</asm.version>
+        <asm.version>9.4</asm.version>
         <bcel.version>6.5.0</bcel.version>
+
+<!--        <maven.compiler.source>11</maven.compiler.source>-->
```
Comment (dkapil) [inline core-java-modules/core-java-jvm/pom.xml:194]: same for here

File: core-java-modules/core-java-jvm/src/main/java/com/baeldung/classloader/PrintClassLoader.java [hunk 3]
```diff
@@ -15,7 +15,7 @@ public class PrintClassLoader {
     public void printClassLoaders() throws ClassNotFoundException {
 
         System.out.println("Classloader of this class:" + PrintClassLoader.class.getClassLoader());
-        System.out.println("Classloader of Logging:" + Logging.class.getClassLoader());
+//        System.out.println("Classloader of Logging:" + Logging.class.getClassLoader());
```
Comment (dkapil) [inline core-java-modules/core-java-jvm/src/main/java/com/baeldung/classloader/PrintClassLoader.java:18]: let's remove commented code from all the files in this PR including java and XML files

File: core-java-modules/core-java-jvm/src/test/java/com/baeldung/bytecode/ViewBytecodeUnitTest.java [hunk 4]
```diff
@@ -43,7 +43,7 @@ public void whenUsingJavassist_thenReadBytecode() throws NotFoundException {
         ClassFile cf = cp.get("java.lang.Object").getClassFile();
         
         assertEquals(cf.getName(), "java.lang.Object");
-        assertEquals(cf.getMethods().size(), 14);
+        assertEquals(cf.getMethods().size(), 12);
```
Comment (dkapil) [inline core-java-modules/core-java-jvm/src/test/java/com/baeldung/bytecode/ViewBytecodeUnitTest.java:46]: is this needed? do we need any article change for this?

File: core-java-modules/core-java-lang-2/src/test/java/com/baeldung/headlessmode/HeadlessModeUnitTest.java [hunk 5]
```diff
@@ -84,4 +83,10 @@ public void whenHeaded_thenFlexibleAppAdjustsItsBehavior() {
         assertThat(FlexibleApp.iAmFlexible()).isEqualTo(FlexibleApp.HEADED);
     }
 
+    private BufferedImage removeAlphaChannel(BufferedImage inputImage) {
```
Comment (dkapil) [inline core-java-modules/core-java-lang-2/src/test/java/com/baeldung/headlessmode/HeadlessModeUnitTest.java:86]: do we need these changes in the article? are these related to JDK upgrade?

File: core-java-modules/pre-jpms/pom.xml [hunk 6]
```diff
@@ -72,8 +80,8 @@
     <properties>
         <dependency.plugin.version>3.1.1</dependency.plugin.version>
         <compiler.plugin.version>3.8.0</compiler.plugin.version>
-        <source.version>1.8</source.version>
-        <target.version>1.8</target.version>
+        <source.version>1.9</source.version>
```
Comment (dkapil) [inline core-java-modules/pre-jpms/pom.xml:83]: let's upgrade to JDK11 directly as we are doing for all other modules

File: pom.xml [hunk 7]
```diff
@@ -374,7 +374,7 @@
                 <module>messaging-modules</module>
 
                 <module>microservices-modules</module>
-                <module>muleesb</module>
+<!--                <module>muleesb</module>-->
```
Comment (dkapil) [inline pom.xml:377]: why are we commenting out this module?

**Review by dkapil — CHANGES_REQUESTED** [review]

File: core-java-modules/core-java-jvm/pom.xml (hunk 1 above)
Comment (anastasiosioannidis) [reply core-java-modules/core-java-jvm/pom.xml:60]: Done

File: core-java-modules/core-java-jvm/pom.xml (hunk 2 above)
Comment (anastasiosioannidis) [reply core-java-modules/core-java-jvm/pom.xml:194]: Done

File: core-java-modules/core-java-jvm/src/main/java/com/baeldung/classloader/PrintClassLoader.java (hunk 3 above)
Comment (anastasiosioannidis) [reply core-java-modules/core-java-jvm/src/main/java/com/baeldung/classloader/PrintClassLoader.java:18]: Done

File: core-java-modules/core-java-jvm/src/test/java/com/baeldung/bytecode/ViewBytecodeUnitTest.java (hunk 4 above)
Comment (anastasiosioannidis) [reply core-java-modules/core-java-jvm/src/test/java/com/baeldung/bytecode/ViewBytecodeUnitTest.java:46]: Yes, it needed with the new Java version.
Yes, the article is: https://www.baeldung.com/java-class-view-bytecode

File: core-java-modules/pre-jpms/pom.xml (hunk 6 above)
Comment (anastasiosioannidis) [reply core-java-modules/pre-jpms/pom.xml:83]: Done

**Comment by anastasiosioannidis** [comment]
retest this please

**Comment by anastasiosioannidis** [comment]
retest this please
